lib/good_job/metrics.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+module GoodJob # :nodoc:
+  class Metrics
+    attr_reader :failed_count, :succeeded_count
+
+    def initialize
+      @failed_count = 0
+      @succeeded_count = 0
+    end
+
+    def increment_failed
+      @failed_count += 1
```

Comment:
`+=` isn't threadsafe. This will need to be Concurrent types (`AtomicFixnum`?)

Reply:
Indeed - fixed in 3111a289b91f67e57077599a4fd66293de64c4ed. Actually, I just used ```CleanupTracker``` as a start - suppose it should be threadsafe as well?